ingredient/ingredient-kubernetes/src/plugin.ts

```diff
@@ -0,0 +1,78 @@
+import fs from 'fs'
+import { BaseIngredient, BakeVariable } from "@azbake/core"
+import { promisify } from 'util';
+import { exec as exec_from_child_process } from 'child_process';
+import Replace from 'replace-in-file';
+const exec = promisify(exec_from_child_process);
+
+export class KubernetesPlugin extends BaseIngredient {
+    
+    public async Execute(): Promise<void> {
+        try {
+            let k8sYamlPath = await this._ingredient.properties.source.valueAsync(this._ctx);            
+            if (!await promisify(fs.exists)(k8sYamlPath)){
+                throw "file/path not found: " + k8sYamlPath
+            }
+            await this.replaceTokens(k8sYamlPath);
+            let testDeployment = this._ingredient.properties.parameters.get("testDeployment");
+            let k8sConfigPath = this._ingredient.properties.parameters.get("config");
+            let configParam = "";
+            if (k8sConfigPath){
+                configParam = `--kubeconfig=${await k8sConfigPath.valueAsync(this._ctx)}`;

```

Comment:
This is still coming in as a file, which doesn't work. Have to pass in the file contents (probably base64 encoded) in this variable. So need to save the contents to a config file, and set --kubeconfig to that file.

Reply:
ah, yeah, onnit